precompiles/avs/tx.go

```diff
@@ -61,24 +69,126 @@ func (p Precompile) RegisterOrDeregisterAVSInfo(
 	return method.Outputs.Pack(true)
 }
 
-func (p Precompile) BindOperatorToAVS(
+func (p Precompile) DeregisterAVS(
 	ctx sdk.Context,
 	_ common.Address,
 	contract *vm.Contract,
 	_ vm.StateDB,
 	method *abi.Method,
 	args []interface{},
 ) ([]byte, error) {
-	if len(args) != 2 {
-		return nil, fmt.Errorf(cmn.ErrInvalidNumberOfArgs, 2, len(args))
+	if len(args) != len(p.ABI.Methods[MethodDeregisterAVS].Inputs) {
+		return nil, xerrors.Errorf(cmn.ErrInvalidNumberOfArgs, len(p.ABI.Methods[MethodDeregisterAVS].Inputs), len(args))
+	}
+	avsParams := &avskeeper.AVSRegisterOrDeregisterParams{}
+
+	avsName, ok := args[0].(string)
+	if !ok || avsName == "" {
+		return nil, xerrors.Errorf(exocmn.ErrContractInputParaOrType, 0, "string", avsName)
+	}
+	avsParams.AvsName = avsName
+
+	avsAddress, err := util.ProcessAddress(contract.Address().String())
+	if err != nil {
+		return nil, errorsmod.Wrap(err, "parse avsAddress error")
+	}
+	avsParams.AvsAddress = avsAddress
+
+	callerAddress, err := util.ProcessAddress(contract.CallerAddress.String())
```

Comment:
Should check if the caller address equal the avs owner address to restrict the permission?

Reply:
Still need to be resolved here, to check if the callerAddress has the authority

Reply:
The permission has been processed in Keepr.go